metal/v1/model_address.go

```diff
@@ -252,6 +252,21 @@ func (o *Address) SetZipCode(v string) {
 	o.ZipCode = v
 }
 
+// AssertRequired throws an error if a required field is nil
+func (o *Address) AssertRequired() error {
+	if IsNil(o.Address) {
```

Comment:
Example of how the template change impacts generated code.  The required fields are checked when unmarshalling JSON to an object.  The validator for a particular type only checks the fields that are specifically required by that type; nested models are checked because they are also being unmarshalled at (roughly speaking) the same time, and _their_ unmarshal function calls _their_ assertion function.  This avoids complications with navigating `oneOf` models, etc..